fastai/callback/hook.py

```diff
@@ -164,20 +178,31 @@ def module_summary(learn, *xb):
     #  thus are not counted inside the summary
     #TODO: find a way to have them counted in param number somehow
     infos = layer_info(learn, *xb)
-    n,bs = 64,find_bs(xb)
+    n,bs = 76,find_bs(xb)
     inp_sz = _print_shapes(apply(lambda x:x.shape, xb), bs)
-    res = f"{learn.model.__class__.__name__} (Input shape: {inp_sz})\n"
-    res += "=" * n + "\n"
-    res += f"{'Layer (type)':<20} {'Output Shape':<20} {'Param #':<10} {'Trainable':<10}\n"
+    res = f"{type(learn.model).__name__} (Input shape: {inp_sz})\n"
     res += "=" * n + "\n"
-    ps,trn_ps = 0,0
+    res += f"{'Layer (type)':<20} {'Output Shape':<20} {'Param #':<10} {'Trainable':<10} {'Group':<10}\n"
+    res += "=" * n
+    ps,trn_ps,j = 0,0,0
     infos = [o for o in infos if o is not None] #see comment in previous cell
-    for typ,np,trn,sz in infos:
+    prev_sz = None
+    for typ,np,trn,sz,grp,chnged in infos:
         if sz is None: continue
-        ps += np
-        if trn: trn_ps += np
-        res += f"{typ:<20} {_print_shapes(sz, bs)[:19]:<20} {np:<10,} {str(trn):<10}\n"
-        res += "_" * n + "\n"
+        if j == 0: res += f'\n{"":<20} {_print_shapes(sz, bs)[:19]:<20}' # to avoid a double line at the top
+        if not chnged and not prev_sz == sz and j > 0: res += "\n" + "_" * n + "\n" + f'{"":<20} {_print_shapes(sz, bs)[:19]:<20}'
+        j = 1
+        if np is '':
+            if sz is not '':  # Pooling layer
+                res += f"\n{typ:<20} {'':<20} {'':<10} {'':<10} {'':<10}"
```

Comment:
Refactor the dupe formatting codes to avoid duplication, and put conditionals on 1 line where you can